Adds ovnk observ with ovs sampling to docs (#86079)
openshift/openshift-docs: modules/nw-ovn-kubernetes-observability.adoc

// Module included in the following assemblies:
//
// * networking/ovn_kubernetes_network_provider/ovn-kubernetes-troubleshooting-sources.adoc

:_mod-docs-content-type: PROCEDURE
[id="nw-ovn-kubernetes-observability_{context}"]
= Checking OVN-Kubernetes network traffic with OVS sampling using the CLI

> OVN-Kubernetes network traffic can be viewed with OVS sampling via the CLI for the following network APIs: `NetworkPolicy`, `AdminNetworkPolicy`, `BaselineNetworkPolicy`, `UserDefinesdNetwork` isolation, multicast ACLs, and egress firewalls. When enabled, you can view network events in the terminal, which helps for packet tracing and debuggability. 

Author: Fixed.

> The scripts for networking events are found in the `/usr/bin/ovnkube-observ` folder of each OVN-Kubernetes node, and is intended for use when the Network Observability Operator is not installed. Administrators can add the `--add-ovs-collect` option to view network traffic across the node, or pass in additional flags to filter result for specific pods. Additional flags can be found in the "OVN-Kubernetes network traffic with OVS sampling flags" section.

Reviewer: I would say it is intended for debugging while netobserv is intended for observabilty and debugging. You can potentially use this script while netobserv is also enabled.

> The scripts for networking events are found in the `/usr/bin/ovnkube-observ` folder of each OVN-Kubernetes node, and is intended for use when the Network Observability Operator is not installed. Administrators can add the `--add-ovs-collect` option to view network traffic across the node, or pass in additional flags to filter result for specific pods. Additional flags can be found in the "OVN-Kubernetes network traffic with OVS sampling flags" section.

Author: I think that I've addressed this properly. I had to change a few things around.

Unlike the Network Observability Operator, which is intended for observability and debugging, checking OVN-Kubernetes network traffic with OVS sampling using the CLI is intended to help with packet tracing and debuggability. It can be used while the Network Observability Operator is installed, however that is not a requirement.

Administrators can add the `--add-ovs-collect` option to view network traffic across the node, or pass in additional flags to filter result for specific pods. Additional flags can be found in the "OVN-Kubernetes network traffic with OVS sampling flags" section.

Use the following procedure to view OVN-Kubernetes network traffic using the CLI.

.Prerequisites

* You are logged in to the cluster as a user with `cluster-admin` privileges.
* You have created a source pod and a destination pod and ran traffic between them.
* You have created at least one of the following network APIs: `NetworkPolicy`, `AdminNetworkPolicy`, `BaselineNetworkPolicy`, `UserDefinesdNetwork` isolation, multicast, or egress firewalls.

.Procedure

> . Edit the `FeatureGate` CR named `cluster` by entering the following command:

Reviewer 2: @stevsmit From my test, if cluster is created by techpreview, `OVNObservability` enabled by default, not require STEP 1 and 2. 
When cluster is normal, yes, we need to enable it. 
BTW I think the better way to enable this feature should use `oc patch --type=merge --patch '{"spec": {"featureSet": "TechPreviewNoUpgrade"}}' featuregate/cluster` cc @npinaeva

// Module included in the following assemblies:
//
// * networking/ovn_kubernetes_network_provider/ovn-kubernetes-troubleshooting-sources.adoc

:_mod-docs-content-type: PROCEDURE
[id="nw-ovn-kubernetes-observability_{context}"]
= Checking OVN-Kubernetes network traffic with OVS sampling using the CLI
:FeatureName: Checking OVN-Kubernetes network traffic with OVS sampling
include::snippets/technology-preview.adoc[]

> OVN-Kubernetes network traffic can be viewed with OVS sampling via the CLI for the following network APIs: `NetworkPolicy`, `AdminNetworkPolicy`, `BaselineNetworkPolicy`, `UserDefinesdNetwork` isolation, `EgressFirewall`, and multicast ACLs. Scripts for these networking events are found in the `/usr/bin/ovnkube-observ` path of each OVN-Kubernetes node.

Reviewer 3: WDYT of busting out the APIs into a list to improve scanability?

> Unlike the Network Observability Operator, which is intended for observability and debugging, checking OVN-Kubernetes network traffic with OVS sampling using the CLI is intended to help with packet tracing and debuggability. It can be used while the Network Observability Operator is installed, however that is not a requirement.

Reviewer 3: s/debugging/debugability

I think you don't lose the intended meaning with _debugging_ and it is a bit simpler.

Reviewer 3: The first sentence is a touch confusing, because at the start, I think you are going to make a contrast between the Network Observabilty Operator and checking OVN-kube network traffic. But I think you are comparing and contrasting, right? Both are useful for debugging, but the Operator is good for observabilty and checking the network traffic is good for packet tracing.

Anyhow, just a minor comment.

Use the following procedure to view OVN-Kubernetes network traffic using the CLI.

.Prerequisites

* You are logged in to the cluster as a user with `cluster-admin` privileges.
* You have created a source pod and a destination pod and ran traffic between them.
* You have created at least one of the following network APIs: `NetworkPolicy`, `AdminNetworkPolicy`, `BaselineNetworkPolicy`, `UserDefinesdNetwork` isolation, multicast, or egress firewalls.

.Procedure

. To enable the `OVNObservability` with OVS sampling feature, enable `TechPreviewNoUpgrade` feature set in the `FeatureGate` CR named `cluster` by entering the following command:
+
[source,terminal]
----
$ oc patch --type=merge --patch '{"spec": {"featureSet": "TechPreviewNoUpgrade"}}' featuregate/cluster
----
+
.Example output
+
[source,terminal]
----
featuregate.config.openshift.io/cluster patched
----

. Confirm that the `OVNObservability` feature is enabled by entering the following command:
+
[source,terminal]
----
$ oc get featuregate cluster -o yaml
----
+
.Example output
+
[source,yaml]
----
  featureGates:
# ...
    enabled:
    - name: OVNObservability
----

> . Obtain a list of the pods inside of the namespace in which you have created one of the relevant network APIs. Note the `NODE` name of the pods, as they are used in the following step.

Reviewer 3: ```suggestion
. Obtain a list of the pods inside of the namespace in which you have created one of the relevant network APIs by entering the following command. Note the `NODE` name of the pods, as they are used in the following step.
```

+
[source,terminal]
----
$ oc get pods -n <namespace> -o wide
----
+
.Example output
+
[source,terminal]
----
NAME              READY   STATUS    RESTARTS   AGE     IP            NODE                                       NOMINATED NODE   READINESS GATES
destination-pod   1/1     Running   0          53s     10.131.0.23   ci-ln-1gqp7b2-72292-bb9dv-worker-a-gtmpc   <none>           <none>
source-pod        1/1     Running   0          56s     10.131.0.22   ci-ln-1gqp7b2-72292-bb9dv-worker-a-gtmpc   <none>           <none>
----

> . Obtain a list of OVN-Kubernetes pods and locate the pod that shares the same `NODE` as the pods from the previous step:

Reviewer 3: ```suggestion
. Obtain a list of OVN-Kubernetes pods and locate the pod that shares the same `NODE` as the pods from the previous step by entering the following command:
```

+
[source,terminal]
----
$ oc get pods -n openshift-ovn-kubernetes -o wide
----
+
.Example output
+
[source,terminal]
----
NAME        
...                             READY   STATUS    RESTARTS      AGE   IP           NODE                                       NOMINATED NODE   
ovnkube-node-jzn5b              8/8     Running   1 (34m ago)   37m   10.0.128.2   ci-ln-1gqp7b2-72292-bb9dv-worker-a-gtmpc   <none>          
...
----

> . Use `oc exec` to open a bash shell inside of the `ovnkube-node` pod by entering the following command:

Reviewer 3: ```suggestion
. Open a bash shell inside of the `ovnkube-node` pod by entering the following command:
```

minimalism nit

+
[source,terminal]
----
$ oc exec -it <pod_name> -n openshift-ovn-kubernetes -- bash
----

. While inside of the `ovnkube-node` pod, you can run the `ovnkube-observ -add-ovs-collector` script to show network events using the OVS collector. For example:
+
[source,terminal]
----
# /usr/bin/ovnkube-observ -add-ovs-collector
----
+
.Example output
+
[source,terminal]
----
...
2024/12/02 19:41:41.327584 OVN-K message: Allowed by default allow from local node policy, direction ingress
2024/12/02 19:41:41.327593 src=10.131.0.2, dst=10.131.0.6

2024/12/02 19:41:41.327692 OVN-K message: Allowed by default allow from local node policy, direction ingress
2024/12/02 19:41:41.327715 src=10.131.0.6, dst=10.131.0.2
...
----

> . Filter the content by type, for example, source pods, by using the `-filter-src-ip` flag with your pod's IP address. For example:

Reviewer 3: ```suggestion
. You can filter the content by type, such as source pods, by entering the following command with the `-filter-src-ip` flag and your pod's IP address. For example:
```
Nit

+
[source,terminal]
----
# /usr/bin/ovnkube-observ -add-ovs-collector -filter-src-ip <pod_ip_address>
----
+
.Example output
+
[source,terminal]
----
...
Found group packets, id 14
2024/12/10 16:27:12.456473 OVN-K message: Allowed by admin network policy allow-egress-group1, direction Egress
2024/12/10 16:27:12.456570 src=10.131.0.22, dst=10.131.0.23

2024/12/10 16:27:14.484421 OVN-K message: Allowed by admin network policy allow-egress-group1, direction Egress
2024/12/10 16:27:14.484428 src=10.131.0.22, dst=10.131.0.23

2024/12/10 16:27:12.457222 OVN-K message: Allowed by network policy test:allow-ingress-from-specific-pod, direction Ingress
2024/12/10 16:27:12.457228 src=10.131.0.22, dst=10.131.0.23

2024/12/10 16:27:12.457288 OVN-K message: Allowed by network policy test:allow-ingress-from-specific-pod, direction Ingress
2024/12/10 16:27:12.457299 src=10.131.0.22, dst=10.131.0.23
...
----
+
For a full list of flags that can be passed in with `/usr/bin/ovnkube-observ`, see "OVN-Kubernetes network traffic with OVS sampling flags".

[id="observability-ovs-sampling-flags_{context}"]

> == OVN-Kubernetes network traffic with OVS sampling flags

Reviewer 3: Procedures are only supposed to have one heading according to the mod docs guide. I think if you just crack this last section off into a reference module you should be good to go.

The following flags are available to view OVN-Kubernetes network traffic by using the CLI. Append these flags to `ovnkube-observ` in your terminal.

> 

Reviewer 3: In my own work, I have included a command syntax example before the table. 

```
.Command syntax
[source,terminal]
----
$ ovnkube-observ <flag>
----
```
I know you have it in the last sentence, but sometimes I find it makes the relationship between the command and the flags a bit more scannable.

just a nit.
